services/libs/aster-std/src/syscall/time.rs

```diff
@@ -5,14 +5,20 @@ use crate::prelude::*;
 use crate::time::SystemTime;
 use crate::util::write_val_to_user;
 
-use super::SyscallReturn;
-use super::SYS_TIME;
+use super::{SyscallReturn, SYS_TIME};
 
 pub fn sys_time(tloc: Vaddr) -> Result<SyscallReturn> {
     log_syscall_entry!(SYS_TIME);
     debug!("tloc = 0x{tloc:x}");
-    let now = SystemTime::now();
-    let now_as_secs = now.duration_since(&SystemTime::UNIX_EPOCH)?.as_secs();
-    write_val_to_user(tloc, &now_as_secs)?;
+
+    let now_as_secs = {
+        let now = SystemTime::now();
+        now.duration_since(&SystemTime::UNIX_EPOCH)?.as_secs()
+    };
+
+    if tloc != 0 {
+        write_val_to_user(tloc, &now_as_secs)?;
```

Comment:
Are you really sure this is the right behaviors? Ignoring an invalid input pointer is a weird design decision. In addition, the man page for the `time` function states that the return value of the function also returns the time, which is not implemented here. So I don't really know what behaviors are correct.

Reply:
The syscall should **always** return the seconds. The problem is fixed now.

Only when the `tloc` addr is not null, the syscall will store the results in the `tloc` address. So the write operation will be skipped if `tloc` is 0.